src/arrayMethodSort.js

```diff
@@ -4,8 +4,45 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const arr = this.slice();
+
+    function swap(array, i, j) {
+      const temp = array[i];
+
+      array[i] = array[j];
+      array[j] = temp;
+    }
+
+    function compare(a, b) {
+      if (compareFunction) {
+        return compareFunction(a, b);
+      } else {
+        if (String(a) < String(b)) {
+          return -1;
+        }
+
+        if (String(a) > String(b)) {
+          return 1;
+        }
+
+        return 0;
+      }
+    }
```

Comment:
```suggestion
```
just set compareFunction as default argument and call compareFunction instead compare in loops